website/docs/vision.md

```diff
@@ -0,0 +1,54 @@
+---
+title: Vision
+sidebar_position: 2
+---
+
+OpenDAL VISION: **access data freely**.
+
+### Free from services
+
+OpenDAL must enable users to access various storage services ranging from `s3` to `dropbox` via it's own native API. It should provide a unified API for accessing all these services.
+
+- Add support for google drive: Good, it allows users to access and manage their data on the google drive.
+- Add support for oss via native API: Good, users can utilize Aliyun's RAM support.
+- Add support for [supabase storage](https://supabase.com/docs/guides/storage): Good, users can visit supabase storage now!
+
+
+- Add support for gcs via XML API: Bad, gcs has native JSON API which more powerful
+- Add support for MySQL/PostgreSQL: Bad, relational DBMS provides data types such as BLOB, but they are often not used as a storage service.
+
+### Free from implementations
+
+OpenDAL needs to separate the various implementation details of services and enables users to write identical logic for different services.
+
+- Add a new capability to indicate whether or not `presign` is supported: Good, users can now write logic based on the `can_presign` option.
+- Add a `default_storage_class` configuration for the S3 service: Good, configuration is specific to the s3 service.
+- Add an option for `content_type` in the `write` operation: Good, it aligns with HTTP standards.
+
+
+- Add a new option in read for `storage_class`: Bad, as different services could have varying values for this parameter.
+
+### Free to integrate
+
+OpenDAL needs to be integrated with different systems.
+
+- Add python binding: Good, users from `python` can use OpenDAL.
+- Add object_store integration: Good, users of `object_store` can adopt OpenDAL.
+
+
+- Rewrite OpenDAL in Golang: Good idea itself, but not related to OpenDAL.
```

Comment:
What does this part mean? This doesn't look like some kind of vision.

Reply:
Examples for whether this idea matches our VISION.

Reply:
May add an explanation of the title of each section, to introduce why OpenDAL can make user become more free.

Reply:
> May add an explanation of the title of each section, to introduce why OpenDAL can make user become more free.

The VISION describes the direction that OpenDAL wants to pursue. It serves as a guide for determining which features should be accepted or rejected. The details of why and how are not included in the VISION.